### ISSUE TITLE
0.6.7 can't find dbt models?

### ISSUE THREAD

**Issue by jdub** [sqlfluff/sqlfluff#1510]
## Expected Behaviour

With 0.6.6 (ignore the failures here, that's a whole other issue I should report):

```
$ sqlfluff lint -v
==== sqlfluff ====
sqlfluff:                0.6.6 python:                  3.9.7
implementation:        cpython dialect:              bigquery
verbosity:                   1 templater:                 dbt
dbt:                   =0.20.2

==== readout ====

=== [ path: /Users/jeffwaugh/src/dbt-swordbill ] ===

=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/accounts/base/base_accounts__access.sql] PASS
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/accounts/base/base_accounts__account_data.sql] PASS
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/accounts/stg_accounts__accounts.sql] FAIL
L:   5 | P:  19 |  PRS | Line 48, Position 1: Found unparsable section: '),
                       | __dbt__cte__base_accounts__access as...'
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/accounts/stg_accounts__profiles.sql] PASS
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/billing/base/base_billing__action.sql] PASS
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/billing/base/base_billing__subscription.sql] PASS
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/billing/stg_billing__actions.sql] FAIL
L:   5 | P:  19 |  PRS | Line 31, Position 1: Found unparsable section:
                       | '),actions as (\n\n select * from __dbt_...'
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/billing/stg_billing__payments.sql] PASS
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/billing/stg_billing__registered_users.sql] PASS
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/billing/stg_billing__subscriptions.sql] FAIL
L:   5 | P:  19 |  PRS | Line 39, Position 1: Found unparsable section:
                       | '),subscriptions as (\n\n select * from ...'
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/catalogue/stg_catalogue__programs.sql] PASS
==== summary ====
violations:        3 status:         FAIL
All Finished 📜 🎉!
```

## Observed Behaviour

With 0.6.7 and no other changes:

```
$ sqlfluff lint -v
==== sqlfluff ====
sqlfluff:                0.6.7 python:                  3.9.7
implementation:        cpython dialect:              bigquery
verbosity:                   1 templater:                 dbt
dbt:                   =0.20.2

==== readout ====

=== [ path: /Users/jeffwaugh/src/dbt-swordbill ] ===

=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
WARNING    Dependent model source.swordbill.accounts.access not found in project. Skipping. 
== [/Users/jeffwaugh/src/dbt-swordbill/models/staging/accounts/base/base_accounts__access.sql] PASS
==== summary ====
violations:        0 status:         PASS
All Finished 📜 🎉!
```

## Steps to Reproduce

I'm… not sure. I can try to construct a simple dbt project that does the same thing.

## Dialect

bigquery

## Version

```
$ python --version
Python 3.9.7
```
```
$ dbt --version
installed version: 0.20.2
...
```
```
$ sqlfluff --version
sqlfluff, version 0.6.7
```

## Configuration

```
[sqlfluff]
dialect = bigquery
templater = dbt

# https://docs.sqlfluff.com/en/stable/rules.html
# L016: Line is too long
# L034: Use wildcards then simple targets before calculations and aggregates in select statements
exclude_rules = L016,L034

[sqlfluff:rules:L014]
extended_capitalisation_policy = lower
```

**Comment by tunetheweb** [comment]
Could you have a look @barrywhart ? Suspect related to #1496

**Comment by wmorin** [comment]
Hello,I also have the same message with sqlfluff 0.6.7
```
$ sqlfluff lint -v
==== sqlfluff ====
sqlfluff:                0.6.7 python:                  3.9.7
implementation:        cpython dialect:             snowflake
verbosity:                   1 templater:                 dbt
dbt:                   =0.20.2

==== readout ====

=== [ path: /Users/jdoe/src/acme/acme-data-dbt ] ===

=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
WARNING    Dependent model source.acme.chargebee.addons not found in project. Skipping.
== [models/prep/chargebee/prep__chargebee_addons.sql] PASS
== [/Users/jdoe/src/acme/acme-data-dbt/models/base/chargebee/base__chargebee_addons.sql] PASS
==== summary ====
violations:        0 status:         PASS
All Finished 📜 🎉!
```

**Comment by barrywhart** [comment]
@jdub, @wmorin 

Hi, I'm the SQLFluff maintainer who fixed an earlier issue (#857). This change likely caused this new behavior. We need more info before we can diagnose or fix this issue.

Can one of you (or anyone) provide a **complete** example that produces this issue?

Please provide:
* `dbt_project.yml`
* `.sqlfluff`
* One or more .sql files

The example needs to be a complete, standalone example that we can run, i.e. it should not depend on connecting to your database.

**Comment by barrywhart** [comment]
Note that the warning itself is not **necessarily** a sign of a problem, but it does indicate that you're trying to lint a model that depends on another model that dbt can't locate. So it _could_ be a concern. Again, an example is vital here. We can't see your dbt projects, and terminal output alone is insufficient for diagnosing, confirming, or fixing issues.

**Comment by barrywhart** [comment]
Ok, I've reproduced the problem. We should have a fix in the next few hours at worst.

**Comment by tunetheweb** [comment]
We have a fix, is anyone able to test this locally just to be 100% sure before we release it, as obviously we missed this before?

**Comment by barrywhart** [comment]
PR is up. Based on advice from a user, I removed the log warning, as it is confusing and not useful.

**Comment by tunetheweb** [comment]
We've just released 0.6.8 which should resolve this. Please let us know if it doesn't (and also if it does - nice to know it's definitely solved!)

**Comment by pecigonzalo** [comment]
I believe this fix re-opened https://github.com/sqlfluff/sqlfluff/issues/857, I have that error now.

**Comment by barrywhart** [comment]
@pecigonzalo: Please create a new issue with your test case. #857 is fixed, and we have an automated test for that now.

Please provide:

* `dbt_project.yml`
* `.sqlfluff`
* One or more .sql files

The example needs to be a complete, standalone example that we can run, i.e. it should not depend on connecting to your database.

**Comment by jdub** [comment]
@barrywhart @tunetheweb Thanks! All while I was asleep! 😆

**Comment by barrywhart** [comment]
Thanks, @jdub. The sun never sets on SQLFluff, ha ha. Is it working okay for you? One user reports they're still seeing the old error about ephemeral models. We're hoping it's working for most people. 😅

**Comment by jdub** [comment]
@barrywhart 0.6.8 is now behaving the way 0.6.6 did, which happens to show errors related to ephemeral models (see "expected behaviour" above), but I don't know if that's the same problem you're talking about. I have to make a smaller, reproducible configuration and file a separate bug about that.

**Comment by pecigonzalo** [comment]
Created in https://github.com/sqlfluff/sqlfluff/issues/1518
